### PR TITLE
Load notebook template through our API.

### PR DESCRIPTION
**Pull request by jimmc**

### CODE REVIEW

File: sources/web/datalab/polymer/modules/template-manager/template-manager.ts [hunk 1]
```diff
@@ -93,6 +100,15 @@ class NotebookTemplate {
   }
 }
 
+// A small hack to allow us to have a unified way of expressing
+// a file location on any of Drive, Jupyter, or our API.
+class ApiFileId extends DatalabFileId {
+  constructor(queryString: string) {
+    const tokens = queryString.split(':');
```
Comment (yebrahim) [inline sources/web/datalab/polymer/modules/template-manager/template-manager.ts]: Throw for bad `queryString` format?

File: sources/web/datalab/polymer/modules/template-manager/template-manager.ts [hunk 2]
```diff
@@ -93,6 +100,15 @@ class NotebookTemplate {
   }
 }
 
+// A small hack to allow us to have a unified way of expressing
+// a file location on any of Drive, Jupyter, or our API.
+class ApiFileId extends DatalabFileId {
```
Comment (yebrahim) [inline sources/web/datalab/polymer/modules/template-manager/template-manager.ts]: I don't understand why we need this. Why not just use a `DatalabFileId`? It'll gladly accept `api` as the source, we just need to add it as a `FileManagerType`.

Comment (jimmc) [reply sources/web/datalab/polymer/modules/template-manager/template-manager.ts]: Done.

File: sources/web/datalab/polymer/modules/template-manager/template-manager.ts (hunk 1 above)
Comment (jimmc) [reply sources/web/datalab/polymer/modules/template-manager/template-manager.ts]: Removed along with the previous change.

**Pull request merged**